tests/workflow_alias_search_attribute_test.go

```diff
@@ -0,0 +1,222 @@
+		TaskQueue: ts.TaskQueue(),
+	}
+	if scheduleByID != "" {
+		scheduleByIDKey := temporal.NewSearchAttributeKeyKeyword("CustomerSA")
+		workflowOptions.TypedSearchAttributes = temporal.NewSearchAttributes(
+			scheduleByIDKey.ValueSet(scheduleByID),
+		)
+	}
+	workflowRun, err := ts.SdkClient().ExecuteWorkflow(ctx, workflowOptions, workflowFn)
+	ts.NoError(err)
+	ts.NotNil(workflowRun)
+
+	return workflowRun
+}
+
+func (ts *WorkflowAliasSearchAttributeTestSuite) TestWorkflowAliasSearchAttribute() {
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	ts.Worker().RegisterWorkflow(ts.WorkflowFunc)
+	ts.Worker().RegisterActivity(ts.ActivityFunc)
+
+	workflowRun1 := ts.createWorkflow(ctx, ts.WorkflowFunc, "wf_id_1", "")
+	workflowRun2 := ts.createWorkflow(ctx, ts.WorkflowFunc, "wf_id_2", "")
+
+	// Wait for workflows to complete
+	err := workflowRun1.Get(ctx, nil)
+	ts.NoError(err)
+	err = workflowRun2.Get(ctx, nil)
+	ts.NoError(err)
+
+	getWFResponse, err := ts.SdkClient().DescribeWorkflowExecution(ctx, workflowRun1.GetID(), workflowRun1.GetRunID())
+	ts.NoError(err)
+	ts.NotNil(getWFResponse)
+	ts.Equal(workflowRun1.GetID(), getWFResponse.GetWorkflowExecutionInfo().Execution.WorkflowId)
+	ts.Equal(workflowRun1.GetRunID(), getWFResponse.GetWorkflowExecutionInfo().Execution.RunId)
+
+	getWFResponse, err = ts.SdkClient().DescribeWorkflowExecution(ctx, workflowRun2.GetID(), workflowRun2.GetRunID())
+	ts.NoError(err)
+	ts.NotNil(getWFResponse)
+	ts.Equal(workflowRun2.GetID(), getWFResponse.GetWorkflowExecutionInfo().Execution.WorkflowId)
+	ts.Equal(workflowRun2.GetRunID(), getWFResponse.GetWorkflowExecutionInfo().Execution.RunId)
+
+	// Use Eventually pattern to wait for visibility store to be updated
+	ts.Eventually(
+		func() bool {
+			resp, err := ts.SdkClient().ListWorkflow(ctx, &workflowservice.ListWorkflowExecutionsRequest{
+				Namespace: ts.Namespace().String(),
+				Query:     "WorkflowId = '" + workflowRun1.GetID() + "' OR WorkflowId = '" + workflowRun2.GetID() + "'",
+			})
+			ts.NoError(err)
+			ts.NotNil(resp)
+			return len(resp.GetExecutions()) == 2
+		},
+		testcore.WaitForESToSettle,
+		100*time.Millisecond,
+	)
+
+	ts.Eventually(
+		func() bool {
+			resp, err := ts.SdkClient().ListWorkflow(ctx, &workflowservice.ListWorkflowExecutionsRequest{
+				Namespace: ts.Namespace().String(),
+				Query:     "TemporalScheduledById IS NULL",
+			})
+			ts.NoError(err)
+			ts.NotNil(resp)
+			return len(resp.GetExecutions()) == 2
+		},
+		testcore.WaitForESToSettle,
+		100*time.Millisecond,
+	)
+
+	ts.Eventually(
+		func() bool {
+			resp, err := ts.SdkClient().ListWorkflow(ctx, &workflowservice.ListWorkflowExecutionsRequest{
+				Namespace: ts.Namespace().String(),
+				Query:     "ScheduledById IS NULL",
+			})
+			ts.NoError(err)
+			ts.NotNil(resp)
+			return len(resp.GetExecutions()) == 2
+		},
+		testcore.WaitForESToSettle,
+		100*time.Millisecond,
+	)
```

Comment:
Not really understanding the intention of this test.
The unit tests are very clear, I would copy that pattern here.